src/spaceone/inventory/service/cloud_service_service.py

```diff
@@ -181,30 +175,35 @@ def update(self, params):
         old_cloud_svc_data = dict(cloud_svc_vo.to_dict())
 
         if 'tags' in params:
-            new_tags = self._create_tag(params, tag_type)
-
-            new_tags_dict: dict = self._change_tags_to_dict(new_tags)
-            old_tags_dict: dict = self._change_tags_to_dict(old_cloud_svc_data['tags'], tag_type=tag_type)
-
-            if new_tags_dict != old_tags_dict:
-                params['tags'] = self._merge_tags(new_tags, old_cloud_svc_data['tags'], tag_type=tag_type)
-            elif 'tags' in params:
+            old_tags = old_cloud_svc_data.get('tags', {})
+            old_tag_keys = old_cloud_svc_data.get('tag_keys', {})
+            new_tags, new_tag_keys = self._convert_tag_type(params['tags'], provider)
+
+            if new_tags[provider] != old_tags.get(provider, {}):
+                old_tags.update(new_tags)
+                old_tag_keys.update(new_tag_keys)
+                params['tags'] = old_tags
+                params['tag_keys'] = list(set(old_tag_keys))
+            else:
                 del params['tags']
 
-        params['collection_info'] = self._get_collection_info(old_cloud_svc_data.get('collection_info', {}))
-
         if 'metadata' in params:
-            params['metadata'] = self._change_metadata_path(params['metadata'])
+            old_metadata = old_cloud_svc_data.get('metadata', {})
+            new_metadata = self._convert_metadata(params['metadata'], provider)
+
+            if new_metadata != old_metadata:
```

Comment:
change condition

`if self._is_different_data(new_xxx, old_xxx, provider):`

Reply:
have reflected thank you.